src/test/java/org/opentripplanner/graph_builder/linking/LinkStopToPlatformTest.java

```diff
@@ -20,77 +23,219 @@
+    AreaEdgeList areaEdgeList = new AreaEdgeList(polygon, Set.of());
 
-    ArrayList<AreaEdge> edges = new ArrayList<>();
+    // visibility vertices are platform entrance points and convex corners
+    // which should be directly linked with stops
+    for (int i : visible) {
+      areaEdgeList.visibilityVertices.add(vertices.get(i));
+    }
 
-    edges.add(createAreaEdge(vertices.get(0), vertices.get(1), areaEdgeList, "edge 1"));
-    edges.add(createAreaEdge(vertices.get(1), vertices.get(2), areaEdgeList, "edge 2"));
-    edges.add(createAreaEdge(vertices.get(2), vertices.get(3), areaEdgeList, "edge 3"));
-    edges.add(createAreaEdge(vertices.get(3), vertices.get(4), areaEdgeList, "edge 4"));
-    edges.add(createAreaEdge(vertices.get(4), vertices.get(0), areaEdgeList, "edge 5"));
+    // AreaEdgeList must include a valid NamedArea which defines area atttributes
+    NamedArea namedArea = new NamedArea();
+    namedArea.setName(new LocalizedString("test platform"));
+    namedArea.setPermission(StreetTraversalPermission.PEDESTRIAN_AND_BICYCLE);
+    namedArea.setOriginalEdges(polygon);
+    areaEdgeList.addArea(namedArea);
 
-    edges.add(createAreaEdge(vertices.get(1), vertices.get(0), areaEdgeList, "edge 6"));
-    edges.add(createAreaEdge(vertices.get(2), vertices.get(1), areaEdgeList, "edge 7"));
-    edges.add(createAreaEdge(vertices.get(3), vertices.get(2), areaEdgeList, "edge 8"));
-    edges.add(createAreaEdge(vertices.get(4), vertices.get(3), areaEdgeList, "edge 9"));
-    edges.add(createAreaEdge(vertices.get(0), vertices.get(4), areaEdgeList, "edge 10"));
+    ArrayList<AreaEdge> edges = new ArrayList<>();
 
-    RegularStop stop = TransitModelForTest
-      .stop("TestStop")
-      .withCoordinate(59.13545, 10.22213)
-      .build();
+    for (int i = 0; i < platform.length; i++) {
+      int next_i = (i + 1) % platform.length;
+      edges.add(createAreaEdge(vertices.get(i), vertices.get(next_i), areaEdgeList, "edge " + i));
+      edges.add(
+        createAreaEdge(
+          vertices.get(next_i),
+          vertices.get(i),
+          areaEdgeList,
+          "edge " + String.valueOf(i + platform.length)
+        )
+      );
+    }
+
+    RegularStop transitStops[] = new RegularStop[stops.length];
+    for (int i = 0; i < stops.length; i++) {
+      Coordinate stop = stops[i];
+      transitStops[i] =
+        TransitModelForTest.stop("TestStop " + i).withCoordinate(stop.y, stop.x).build();
+    }
 
     transitModel.index();
     graph.index(transitModel.getStopModel());
 
-    new TransitStopVertexBuilder().withGraph(graph).withStop(stop).build();
+    for (RegularStop s : transitStops) {
+      new TransitStopVertexBuilder().withGraph(graph).withStop(s).build();
+    }
+
+    return graph;
+  }
+
+  /**
+   * Link stop outside platform area to platform.
+   */
+  @Test
+  public void testLinkStopOutsideArea() {
+    // test platform is a simple rectangle. It creates a graph of 8 edges.
+    Coordinate platform[] = {
```

Comment:
```suggestion
    Coordinate[] platform = {
```

The above line is not wrong but unusual for Java. There are a few others also in the file.

Reply:
right! Fixed now.